app/src/main/java/org/hisp/dhis/android/dataentry/reports/EnrollmentsRepositoryImpl.java

```diff
@@ -0,0 +1,137 @@
+
+import com.squareup.sqlbrite.BriteDatabase;
+
+import org.hisp.dhis.android.core.common.State;
+import org.hisp.dhis.android.core.enrollment.EnrollmentModel;
+import org.hisp.dhis.android.dataentry.commons.tuples.Pair;
+import org.hisp.dhis.android.dataentry.commons.tuples.Sextet;
+
+import java.util.List;
+import java.util.Locale;
+
+import io.reactivex.Flowable;
+
+import static hu.akarnokd.rxjava.interop.RxJavaInterop.toV2Flowable;
+import static org.hisp.dhis.android.dataentry.commons.utils.DbUtils.string;
+
+@SuppressWarnings({
+        "PMD.AvoidDuplicateLiterals"
+})
+final class EnrollmentsRepositoryImpl implements ReportsRepository {
+    private static final String SELECT_ENROLLMENTS = "SELECT" +
+            "  Program.displayName," +
+            "  Program.enrollmentDateLabel," +
+            "  Enrollment.enrollment," +
+            "  Enrollment.enrollmentDate," +
+            "  Enrollment.state," +
+            "  Enrollment.status," +
+            "  InstanceAttribute.label," +
+            "  TrackedEntityAttributeValue.value " +
+            "FROM (Enrollment INNER JOIN Program ON Program.uid = Enrollment.program)" +
+            "  LEFT OUTER JOIN (" +
+            "    SELECT TrackedEntityAttribute.uid AS tea, " +
+            "           TrackedEntityAttribute.displayName AS label, " +
+            "           ProgramTrackedEntityAttribute.displayInList AS showInList, " +
+            "           ProgramTrackedEntityAttribute.program AS program, " +
+            "           ProgramTrackedEntityAttribute.sortOrder AS formOrder " +
+            "    FROM ProgramTrackedEntityAttribute INNER JOIN TrackedEntityAttribute" +
+            "      ON TrackedEntityAttribute.uid = ProgramTrackedEntityAttribute.trackedEntityAttribute" +
+            "    ) AS InstanceAttribute ON InstanceAttribute.program = Program.uid " +
+            "        AND InstanceAttribute.showInList = 1" +
+            "  LEFT OUTER JOIN TrackedEntityAttributeValue" +
+            "      ON (TrackedEntityAttributeValue.trackedEntityAttribute = InstanceAttribute.tea " +
+            "        AND TrackedEntityAttributeValue.trackedEntityInstance = Enrollment.trackedEntityInstance) " +
+            "WHERE Enrollment.trackedEntityInstance = ? AND NOT Enrollment.state = 'TO_DELETE' " +
+            "ORDER BY datetime(Enrollment.created) DESC, " +
+            "Enrollment.enrollment ASC, InstanceAttribute.formOrder ASC;";
+
+    @NonNull
+    private final BriteDatabase briteDatabase;
+
+    @NonNull
+    private final String teiUid;
+
+    @NonNull
+    private final String promptProgram;
+
+    @NonNull
+    private final String promptEnrollmentStatus;
+
+    @NonNull
+    private final String promptEnrollmentDateLabel;
+
+    EnrollmentsRepositoryImpl(@NonNull BriteDatabase briteDatabase, @NonNull String teiUid,
+            @NonNull String promptProgram, @NonNull String promptEnrollmentStatus,
+            @NonNull String promptEnrollmentDateLabel) {
```

Comment:
I solved getting the Uids / prompts in a different way: Storing them in the View (i.e. fragment) and then exposing them to the Repository. Example for this class `public Flowable<List<ReportViewModel>> reports(String teiUid)`.
And then in ReportsPresenter you do `repository.reports(reportsView.tei());`